### ISSUE TITLE
Alliance victory rework

### ISSUE THREAD

**Issue by subchannel13** [iktools/Ancient-Star#49]
The problem with how alliance victory works now is that getting the condition (all players allied with all other players) is less likely with more players in the game. The proposition is to relax the condition to at least one player allied to all others but refuse forming the alliance if either party is at war with an ally of the other party. This also requires treating relations as a stateful objects, not just pure functions of relation proposals.

- [x] Stateful relations
- [x] New alliance formation condition
- [x] Change alliance victory condition
- [X] Save upgrade
- [x] Allow players to see other player relations with 3rd parties
- [x] Update design document
- [x] Update bot behavior

**Comment by spikerobinson** [comment]
That was quick.
Does the player have to worry about bots forming an alliance and winning against the player?

For bots where does the incentive to come from to not just immediately join a winning alliance (or try to form one)?

I suggest there should be some kind of scoring where (eg) being in an alliance of 3 is worth more to a bot than being in an alliance of 8 - the worst of all scoring being if the bot thinks it is going to be wiped out if it doesn't join some alliance.

Presumably for alliance Victory the winning alliance must destroy all other entities? Or persuade them to join? I guess the last entity to join a hostile alliance is in effect "Surrendering"?

**Comment by spikerobinson** [comment]
Also in eg Uciana you need to consider intra-alliance competion which is the non-combat race to exploit new planets. Players are good at this while bots tend to suck, making the alliance exploitative for them.

**Comment by subchannel13** [comment]
Quick? Not from my perspective, I dragged it for like 3 weeks.

> Does the player have to worry about bots forming an alliance and winning against the player?

Yes, but. Big alliance ends the game, it doesn't make anyone win the game. At the end everyone if "judged" by their score. And only true losers are those who got eliminated.

> I suggest there should be some kind of scoring where (eg) being in an alliance of 3 is worth more to a bot than being in an alliance of 8

That's an interesting thought, I'll keep it in mind (in written form) in case big alliances really end up needing discouragement. As the game is now, large alliances in should not be as appealing because they lock your ranking in place. For the top player that's ok and they'd want to end the game as soon as possible. Other players should try to seek opportunities for improving their ranking with strategic wars and alliances.

> Players are good at this while bots tend to suck, making the alliance exploitative for them.

Bots in the Ancient Star are a lot simpler then players give them credit. I tried some prisoner dilemma math on paper and I have some notes on how to improve bot logic but that will have to wait until I clear higher priority improvements.
